app/src/main/res/layout/fragment_profile_badges.xml

```diff
@@ -1,151 +1,141 @@
 <?xml version="1.0" encoding="utf-8"?>
-<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```

Comment:
Does this need to be a constraint layout? No constraints are used in the layout

Reply:
Earlier issue with badges alignment cut towards right happened likely due to the fixed width and margins in Linear Layout components. This could be the issue for few devices with different screen sizes and resolutions. So, Instead of using fixed widths and margins, and to make it more responsive for any type of device used relative positioning and constraints through Constraint Layout